internal/storage/storage_handle.go

```diff
@@ -42,57 +38,30 @@ type storageClient struct {
 	client *storage.Client
 }
 
-type StorageClientConfig struct {
-	ClientProtocol      mountpkg.ClientProtocol
-	MaxConnsPerHost     int
-	MaxIdleConnsPerHost int
-	TokenSrc            oauth2.TokenSource
-	HttpClientTimeout   time.Duration
-	MaxRetryDuration    time.Duration
-	RetryMultiplier     float64
-	UserAgent           string
-}
-
 // NewStorageHandle returns the handle of Go storage client containing
 // customized http client. We can configure the http client using the
 // storageClientConfig parameter.
-func NewStorageHandle(ctx context.Context, clientConfig StorageClientConfig) (sh StorageHandle, err error) {
-	var transport *http.Transport
-	// Using http1 makes the client more performant.
-	if clientConfig.ClientProtocol == mountpkg.HTTP1 {
-		transport = &http.Transport{
-			MaxConnsPerHost:     clientConfig.MaxConnsPerHost,
-			MaxIdleConnsPerHost: clientConfig.MaxIdleConnsPerHost,
-			// This disables HTTP/2 in transport.
-			TLSNextProto: make(
-				map[string]func(string, *tls.Conn) http.RoundTripper,
-			),
-		}
-	} else {
-		// For http2, change in MaxConnsPerHost doesn't affect the performance.
-		transport = &http.Transport{
-			DisableKeepAlives: true,
-			MaxConnsPerHost:   clientConfig.MaxConnsPerHost,
-			ForceAttemptHTTP2: true,
+func NewStorageHandle(ctx context.Context, clientConfig storageutil.StorageClientConfig) (sh StorageHandle, err error) {
+
+	var clientOpts []option.ClientOption
+
+	// Add WithHttpClient option.
+	if clientConfig.ClientProtocol == mountpkg.HTTP1 || clientConfig.ClientProtocol == mountpkg.HTTP2 {
+		httpClient, clientErr := storageutil.CreateHttpClientObj(&clientConfig)
```

Comment:
why not use err directly here. why do you need seperate variable.

Reply:
Yeah, we can do like this:
```go
var httpClient *http.Client
httpClient, err = storageutil.CreateHttpClientObj(&clientConfig)
if err != nil {
	err = fmt.Errorf("while creating http endpoint: %w", err)
	return
}
```

Other we use `:` in that case, it gives the below error: 
```internal/storage/storage_handle.go:54:4: result parameter err not in scope at return
        internal/storage/storage_handle.go:51:15: inner declaration of var err error
```